src/Serialization/containers.jl

```diff
@@ -145,6 +188,52 @@ function load_object(s::DeserializerState, T::Type{<:Matrix{S}}, params::NCRing)
   end
 end
 
+function load_object(s::DeserializerState, T::Type{<:Matrix{S}}, key::U) where {S, U <: Union{Int, Symbol}}
+  return load_node(s, key) do _
+    load_object(s, T)
+  end
+end
+
+
+################################################################################
+# Multidimensional Arrays
+@register_serialization_type Array "MultiDimArray"
+
+function save_object(s::SerializerState, arr::Array{T, N}) where {T, N}
+  arr_size = size(arr)
+  k = arr_size[N]
+  save_data_array(s) do
+    for t in 1:k
+      save_object(s, arr[[1:arr_size[i] for i in 1:N - 1]..., t])
+    end
+  end
+end
+
+function load_object(s::DeserializerState, T::Type{Array{S, N}}) where {S, N}
+  load_node(s) do entries
+    if isempty(entries)
+      return T(undef, [0 for _ in 1:N]...)
+    end
+    len = length(entries)
+    m = [load_object(s, Array{S, N - 1}, i) for i in 1:len]
+    return T(reduce((a, b) -> cat(a, b; dims=N), m))
```

Comment:
same here. This reduction creates a lot of unnecessary allocations. you should be able to create an undef Array of the correct size and then deserialize entry by entry and put them at the correct indices